android/src/main/java/com/revenuecat/purchases/hybridcommon/mappers/StoreProductMapper.kt

```diff
@@ -123,4 +126,36 @@ private fun Period.mapPeriod(): Map<String, Any?>? {
             "periodNumberOfUnits" to 0
         )
     }
+}
+
+private fun SubscriptionOption.mapSubscriptionOption(storeProduct: StoreProduct): Map<String, Any?> {
+    return mapOf(
+        "id" to id,
+        "storeProductId" to storeProduct.id,
+        "productId" to storeProduct.purchasingData.productId,
```

Comment:
what's the difference bt storeProductId and productId?

Reply:
This is... confusing a little bit 🤦‍♂️ And its documented in the Flutter model but it should also be documented here

- `id` is `{basePlanId}:{offerId}` which is used for matching `SubscriptionOption` on purchase
- `storeProductId` is `{subId}:{basePlanId}` is **NOT** used for any purchasing marching because it is done with `id` and `productId`
- `productId` is `{subId}` which is used for querying `getProductsWith(productId)` for only the product we need to find

It's not very clear so I should comment the use case of them here